### PR TITLE
Fixed transformers version conflict and fixed UT error

### PR DESCRIPTION
**Pull request by PenghuiCheng**
# What does this PR do?
Fixed transformers version conflict and fixed UT error

### CODE REVIEW

**Comment by chensuyue** [comment]
https://github.com/huggingface/optimum-intel/actions/runs/7097503652/job/19317724224?pr=483#step:5:747 This case failed due to HF connect issue, @echarlaix could you help to check?

File: tests/neural_compressor/test_modeling.py [hunk 1]
```diff
@@ -161,7 +161,7 @@ def test_compare_with_and_without_past_key_values(self):
         self.assertEqual(outputs_model_with_pkv.shape[1], self.GENERATION_LENGTH)
         self.assertEqual(outputs_model_without_pkv.shape[1], self.GENERATION_LENGTH)
         self.assertTrue(
-            without_pkv_timer.elapsed / with_pkv_timer.elapsed > self.SPEEDUP_CACHE,
+            with_pkv_timer.elapsed / without_pkv_timer.elapsed > self.SPEEDUP_CACHE,
```
[the remark this answers is not in the document]
Comment (PenghuiCheng) [reply tests/neural_compressor/test_modeling.py:164]: So maybe I have a misunderstanding. But it is strange why with_pkv_timer.elapsed > without_pkv_timer.elapsed.

**Comment by echarlaix** [comment]
Could you rebase @PenghuiCheng ? (recent fixes were integrated for the failing tests)

File: tests/neural_compressor/test_modeling.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (PenghuiCheng) [reply tests/neural_compressor/test_modeling.py]: > Not sure to understand this modification : the latency with `use_cache=False` (without leveraging the pre-computed pkv) should be higher than when `use_cache=True` (when the already computed pkv are re-used)

Yes, You are right, now it be passed in UT.

File: setup.py [hunk 2]
```diff
@@ -42,6 +42,7 @@
         "onnx",
         "onnxruntime<1.15.0",
         "transformers>=4.33.0",
+        "intel-extension-for-pytorch",
```
Comment (echarlaix) [inline setup.py:45]: why is it needed ? shouldn't it be optional ?

File: tests/neural_compressor/test_optimization.py [hunk 3]
```diff
@@ -77,7 +77,7 @@ class OptimizationTest(INCTestMixin):
     )
 
     TEXT_GENERATION_SUPPORTED_ARCHITECTURES = (
-        "hf-internal-testing/tiny-random-BloomForCausalLM",
+        "hf-tiny-model-private/tiny-random-BloomForCausalLM",
```
Comment (echarlaix) [inline tests/neural_compressor/test_optimization.py:80]: why this modification ?

Comment (PenghuiCheng) [reply tests/neural_compressor/test_optimization.py:80]: there is error in test_quantize_text_generate_model UT with "hf-internal-testing/tiny-random-BloomForCausalLM" :
2023-12-19 11:14:01 [ERROR] Unexpected exception RuntimeError('INDICES element is out of DATA bounds, id=6168 axis_dim=1024') happened during tunin
g.
Traceback (most recent call last):
  File "/home/penghuic/inc_pr/neural_compressor/quantization.py", line 234, in fit
    strategy.traverse()
  File "/home/penghuic/inc_pr/neural_compressor/strategy/auto.py", line 140, in traverse
    super().traverse()
  File "/home/penghuic/inc_pr/neural_compressor/strategy/strategy.py", line 504, in traverse
    q_model = self.adaptor.quantize(copy.deepcopy(tune_cfg), self.model, self.calib_dataloader, self.q_func)
  File "/home/penghuic/inc_pr/neural_compressor/utils/utility.py", line 304, in fi
    res = func(*args, **kwargs)
  File "/home/penghuic/inc_pr/neural_compressor/adaptor/pytorch.py", line 3630, in quantize
    q_func(q_model._model)
  File "/home/penghuic/optimum-intel-new/tests/neural_compressor/test_optimization.py", line 368, in calibration_fn
    tmp_model.generate(**tokens, max_new_tokens=32, do_sample=False)
  File "/home/penghuic/anaconda3/envs/py3.10/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)

File: setup.py (hunk 2 above)
Comment (PenghuiCheng) [reply setup.py]: Yes, I move it to be optional, This change is only for automated weekly testing.

**Comment by PenghuiCheng** [comment]
hi, Ella, the pre-ci tests on the OpenVino section. It is not the cause of this PR. Could you review this PR?

File: tests/neural_compressor/test_optimization.py (hunk 3 above)
Comment (echarlaix) [reply tests/neural_compressor/test_optimization.py:80]: thanks for your answer @PenghuiCheng, could you let me know in which case this issue is appearing and if yes what is causing it? I don't see it in the current CI, is this appearing when installing `neural-compressor` from source ?

Comment (echarlaix) [reply tests/neural_compressor/test_optimization.py:80]: any news on this @PenghuiCheng ?

Comment (PenghuiCheng) [reply tests/neural_compressor/test_optimization.py:80]: Hi, Ella, Yes, This issue does not exist, it seems that my model has not been fully downloaded. I will close it. thanks!

**Pull request closed without merging**